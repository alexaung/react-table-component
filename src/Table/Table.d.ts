import * as React from 'react';

export interface TableProps {
  title?: string;
  columns?: string[];
  dataRows?: string[][];
  mode?: string;
  maxRowsToDisplay?: number;
  enableToggleMode?: boolean;
  header?: JSX.Element;
  footer?: JSX.Element;
  onChange?: (e: Event) => void;
  [key: string]: any;
}

interface TableState {}

declare class Table extends React.Component<TableProps, TableState> {
  render(): JSX.Element;
}

export default Table;
