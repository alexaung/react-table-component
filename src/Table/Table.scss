@import "../variables/variables.scss";
/* Desktop Styles */
$width: 768px;
@media screen and (min-width: $width) {
  .table-title {
    margin-bottom: 10px;
    font-size: 16px;
    color: $color-dark;
    line-height: 20px;
    letter-spacing: normal;
    font-weight: 600;
  }
  
  .table-wrapper {
    overflow: auto;
  }
  
  .base-table {
    display: table;
    width: 100%;
    border-collapse: collapse;
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.05);
    font-family: $avenirheavy;
    font-weight: 900;
    letter-spacing: 0;
    line-height: 1.75;
    border-radius: 16px;
    thead {
      tr {
        border-bottom: 1px solid $table-row-border-bottom-color;
        background-color: $table-header-color;
        padding: 24px;
        th {
          font-size: 20px;
          color: $color-dark;
          text-align: left;
          letter-spacing: 0.08px;
          font-weight: 600;
          padding: 15px 30px 15px 5px;
          white-space: nowrap;
        }
      }
    }
    tbody {
      tr {
        &:not(:last-child) {
          border-bottom: 1px solid $table-row-border-bottom-color;
        }
        td {
          font-size: 16px;
          color: $color-dark;
          text-align: left;
          letter-spacing: 0.08px;
          font-weight: normal;
          padding: 15px 30px 15px 5px;
          white-space: nowrap;
        }
      }
    }
  }
}
@media screen and (max-width: $width) {
  .table-title {
    margin-bottom: 10px;
    font-size: 16px;
    color: $color-dark;
    line-height: 20px;
    letter-spacing: normal;
    font-weight: 600;
  }
  
  .table-wrapper {
    overflow: auto;
  }
  
  .base-table {
    display: table;
    width: 100%;
    border-collapse: collapse;
    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.05);
    font-family: $avenirheavy;
    font-weight: 900;
    letter-spacing: 0;
    line-height: 1.75;
    border-radius: 16px;
    thead {
      tr {
        display: none;
      }
    }
    tbody {
      tr {
        flex-wrap: wrap;
        padding: 8px 16px;
        &:not(:last-child) {
          margin-bottom: 8px;
          border-bottom: 1px solid #E1E1E1;
        }
        td {
          font-size: 16px;
          color: $color-dark;
          text-align: left;
          letter-spacing: 0.08px;
          font-weight: normal;
          padding: 15px 30px 15px 5px;
        }
      }
    }
  }
}

.enable-table-hover {
  tr:hover {
    background-color: $color-sky-blue;
    cursor: pointer;
  }
}

.selected-row-highlight {
  background-color: $selected-row-color;
}

.input-selection {
  padding: 10px;
  accent-color: $input-selection-color;
  width: 24px;
  height: 24px;
}
