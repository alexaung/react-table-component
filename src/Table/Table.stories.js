import React from 'react';
import { storiesOf } from '@storybook/react';
import { jsxDecorator } from 'storybook-addon-jsx';
import Table from '.';

const headerLabels = [
  'Claim Number',
  'Date of Admission',
  'Date of Discharge',
  'Clinic/Hospital'
];
const dataRows = [
  ['CBGDC21000386-00', '12-03-2021', '26-03-2021', 'PARKWAY EAST'],
  ['CBGDC21000386-00', '13-03-2021', '25-03-2021', 'RESTRUCTURED HOSPITAL'],
  ['CBGDC21000386-00', '14-03-2021', '24-03-2021', 'PARKWAY EAST'],
  ['CBGDC21000386-00', '15-03-2021', '23-03-2021', 'SUNWAY'],
  ['CBGDC21000386-00', '16-03-2021', '22-03-2021', 'KEPPAL'],
  ['CBGDC21000386-00', '17-03-2021', '21-03-2021', 'PARKWAY EAST'],
  ['CBGDC21000386-00', '18-03-2021', '20-03-2021', 'PARKWAY EAST']
];

function Footer() {
  return (
    <div>
      <span>Hi this is footer</span>
    </div>
  );
}

function Header() {
  return (
    <div>
      <span>Hi this is header</span>
    </div>
  );
}

storiesOf('Table|Default', module)
  .addDecorator(jsxDecorator)
  .addParameters({
    info: {
      text: 'Table documentation'
    }
  })
  .add('Basic Table', () => (
    <div style={{ margin: '30px' }}>
      <Table
        title="If YES, Choose Related Hospitalisation/ Day Surgery Claims"
        headerLabels={headerLabels}
        dataRows={dataRows}
      />
    </div>
  ))
  .add('Single selection', () => (
    <div style={{ margin: '30px' }}>
      <Table
        title="If YES, Choose Related Hospitalisation/ Day Surgery Claims"
        headerLabels={headerLabels}
        dataRows={dataRows}
        mode="single"
        onChange={data => {
          console.log('Row data', data);
        }}
      />
    </div>
  ))
  .add('Multiple selection', () => (
    <div style={{ margin: '30px' }}>
      <Table
        title="If YES, Choose Related Hospitalisation/ Day Surgery Claims"
        headerLabels={headerLabels}
        dataRows={dataRows}
        mode="multiple"
        onChange={data => {
          console.log('Row data', data);
        }}
      />
    </div>
  ))
  .add('Limit rows', () => (
    <div style={{ margin: '30px' }}>
      <Table
        title="If YES, Choose Related Hospitalisation/ Day Surgery Claims"
        headerLabels={headerLabels}
        dataRows={dataRows}
        maxRowsToDisplay={3}
        onChange={data => {
          console.log('Row data', data);
        }}
      />
    </div>
  ))
  .add('With footer', () => (
    <div style={{ margin: '30px' }}>
      <Table
        title="If YES, Choose Related Hospitalisation/ Day Surgery Claims"
        headerLabels={headerLabels}
        dataRows={dataRows}
        footer={<Footer />}
      />
    </div>
  ))
  .add('With header', () => (
    <div style={{ margin: '30px' }}>
      <Table
        header={<Header />}
        headerLabels={headerLabels}
        dataRows={dataRows}
        footer={<Footer />}
      />
    </div>
  ))
  .add('Disable toggle', () => (
    <div style={{ margin: '30px' }}>
      <Table
        title="If YES, Choose Related Hospitalisation/ Day Surgery Claims"
        headerLabels={headerLabels}
        dataRows={dataRows}
        mode="single"
        enableToggleMode={false}
        onChange={data => {
          console.log('Row data', data);
        }}
      />
    </div>
  ))
  .add('Compact view', () => (
    <div style={{ margin: '30px', width: '320px' }}>
      <Table
        title="Test header"
        headerLabels={headerLabels}
        dataRows={dataRows}
      />
    </div>
  ));
