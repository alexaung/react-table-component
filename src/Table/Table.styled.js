import styled from "styled-components";

const smwidth = "576px";
const mdwidth = "768px";
const xlwidth = "992px";

const TableWrapper = styled.div`
  @media screen and (min-width: ${xlwidth}) {
    overflow: auto;
    border-radius: 16px;
    box-shadow: ${({ theme }) => theme.tableBoxShadow};
  }

  @media screen and (min-width: ${mdwidth}) {
    overflow: auto;
    border-radius: 16px;
    box-shadow: ${({ theme }) => theme.tableBoxShadow};
  }

  @media screen and (max-width: ${smwidth}) {
    overflow: auto;
    border-radius: 8px;
    box-shadow: ${({ theme }) => theme.tableBoxShadow};
  }
`;

const TableTitle = styled.h1`
  @media screen and (min-width: ${xlwidth}) {
    margin-bottom: 10px;
    font-size: 20px;
    color: ${({ theme }) => theme.primary};
    line-height: 20px;
    letter-spacing: normal;
    font-weight: 600;
  }

  @media screen and (min-width: ${mdwidth}) {
    margin-bottom: 10px;
    font-size: 20px;
    color: ${({ theme }) => theme.primary};
    line-height: 20px;
    letter-spacing: normal;
    font-weight: 600;
  }

  @media screen and (max-width: ${smwidth}) {
    margin-bottom: 10px;
    font-size: 20px;
    color: ${({ theme }) => theme.primary};
    line-height: 20px;
    letter-spacing: normal;
    font-weight: 600;
  }
`;

const BaseTable = styled.table`
  @media screen and (min-width: ${xlwidth}) {
    display: table;
    width: 100%;
    border-collapse: collapse;
    line-height: 1.75;
    letter-spacing: 0;

    thead tr {
      background-color: ${({ theme }) => theme.tableHeaderColor};
      cursor: pointer;

      th {
        font-family: ${({ theme }) => theme.avenirHeavy};
        color: ${({ theme }) => theme.primary};
        font-size: 20px;
        text-align: left;
        letter-spacing: 0px;
        padding: 24px;
      }
    }

    tbody tr {
      &:not(:last-child) {
        border-bottom: 1px solid
          ${({ theme }) => theme.tableRowBorderBottomColor};
      }

      td {
        font-family: ${({ theme }) => theme.avenirBook};
        color: ${({ theme }) => theme.text};
        font-size: 20px;
        text-align: left;
        letter-spacing: 0.1px;
        padding: 24px;
      }
    }
  }

  @media screen and (min-width: ${mdwidth}) {
    display: table;
    width: 100%;
    border-collapse: collapse;
    line-height: 1.75;
    letter-spacing: 0;

    thead tr {
      background-color: ${({ theme }) => theme.tableHeaderColor};
      cursor: pointer;

      th {
        font-family: ${({ theme }) => theme.avenirHeavy};
        color: ${({ theme }) => theme.primary};
        font-size: 18px;
        text-align: left;
        letter-spacing: 0px;
        padding: 24px;
      }
    }

    tbody tr {
      &:not(:last-child) {
        border-bottom: 1px solid
          ${({ theme }) => theme.tableRowBorderBottomColor};
      }

      td {
        font-family: ${({ theme }) => theme.avenirBook};
        color: ${({ theme }) => theme.text};
        font-size: 18px;
        text-align: left;
        letter-spacing: 0.1px;
        padding: 24px;
      }
    }
  }

  @media screen and (max-width: ${smwidth}) {
    display: table;
    width: 100%;
    border-collapse: collapse;
    line-height: 1.75;
    letter-spacing: 0;

    thead tr {
      background-color: ${({ theme }) => theme.primary};
      cursor: pointer;

      th {
        font-family: ${({ theme }) => theme.avenirHeavy};
        color: ${({ theme }) => theme.text};
        font-size: 16px;
        text-align: left;
        letter-spacing: 0px;
        padding: 10px;
      }
    }

    tbody tr {
      &:not(:last-child) {
        margin-bottom: 8px;
        border-bottom: 1px solid
          ${({ theme }) => theme.tableRowBorderBottomColor};
      }

      td {
        font-family: ${({ theme }) => theme.avenirBook};
        color: ${({ theme }) => theme.text};
        font-size: 16px;
        text-align: left;
        letter-spacing: 0.1px;
        padding: 10px;
      }
    }
  }
`;

const TBody = styled.tbody`
  tr:hover {
    background-color: ${({ theme, enableTableHover }) =>
      enableTableHover === "enable-table-hover" ? theme.secondary : ""};
    cursor: pointer;
  }
`;

// const SelectedRowHighlight = styled.tr`
//   background-color: ${({ theme }) => theme.selectedRowColor};
// `;

const TableRow = styled.tr`
  background-color: ${({ theme, selected }) =>
    selected ? theme.selectedRowColor : ""};
`;

const Input = styled.input`
  cursor: pointer;
  padding: 10px;
  accent-color: ${({ theme }) => theme.inputColor};
  width: 24px;
  height: 24px;
`;

const TableHeaderCell = styled.div`
  color: ${({ theme, sortDirection }) =>
    sortDirection === "ascending" || sortDirection === "descending"
      ? theme.inputColor
      : ""};
  display: inline-flex;
  align-self: center;
  padding-left: 10px;
`;

export {
  TableWrapper,
  TableTitle,
  BaseTable,
  TBody,
  TableRow,
  Input,
  TableHeaderCell,
};
