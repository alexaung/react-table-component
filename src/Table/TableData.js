import React, { useState, useEffect } from "react";
import { generateKey } from "./../util/utils";

function TableData(props) {
  const { headerLabels, rowData, index } = props;
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    setIsMobile(window.innerWidth <= 700);
  }, []);

  if (isMobile) {
    return (
      <td key={generateKey("td", rowData.join(), index)}>
        {Array.isArray(rowData) &&
          rowData.map((columnData, index) => (
            <p key={index}>
              {headerLabels[index]}: {columnData}
            </p>
          ))}
      </td>
    );
  } else {
    return rowData.map((cell, cellIndex) => <td key={cellIndex}>{cell}</td>);
  }
}

export default TableData;
