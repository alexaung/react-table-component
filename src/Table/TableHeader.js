import React from "react";
import { generateKey } from "./../util/utils";

function TableHeader(props) {
  const { headerLabels, mode } = props;

  return (
    <thead>
      <tr>
      {(mode === 'single' || mode === 'multiple') ? (
        <th></th>
      ) : null}
        
        {Array.isArray(headerLabels) &&
          headerLabels.map((header, index) => (
            <th scope="col" key={generateKey("th", header, index)}>
              {header}
            </th>
          ))}
      </tr>
    </thead>
  );
}

export default TableHeader;
