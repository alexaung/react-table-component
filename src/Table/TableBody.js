import React, { useState, useEffect } from "react";
import { generateKey } from "./../util/utils";
import TableStyles from "./Table.scss";
import classNames from "classnames/bind";
import TableData from "./TableData";

const styles = classNames.bind(TableStyles);
const modes = ["SINGLE", "MULTIPLE"];

function TableBody(props) {
  const { headerLabels, dataRows, mode, onChange, enableToggleMode } = props;
  const [updatedMode, setUpdatedMode] = useState("NONE");
  const [selectionMode, setSelectionMode] = useState("");
  const [selectedRowsKeys, setSelectedRowsKeys] = useState([]);
  const [selectedRowsData, setSelectedRowsData] = useState([]);

  useEffect(() => {
    const updatedMode = mode ? `${mode}`.toUpperCase() : "NONE";
    setUpdatedMode(updatedMode);
    if (modes.includes(updatedMode)) {
      setSelectionMode("enable-table-hover");
    }
  }, [mode]);

  function getSingleSelection(key, rowData) {
    if (enableToggleMode && selectedRowsKeys.includes(key)) {
      setSelectedRowsKeys([]);
      return [];
    }
    setSelectedRowsKeys([key]);
    return rowData;
  }

  function getMultipleSelectionToggleData(key, rowData) {
    const data = selectedRowsData.filter(
      (selectedRowData) =>
        !selectedRowData.every((selectedData) => rowData.includes(selectedData))
    );
    setSelectedRowsData(data);
    setSelectedRowsKeys(
      selectedRowsKeys.filter(
        (selectedRowKey) => ![key].includes(selectedRowKey)
      )
    );
    return data;
  }

  function getMultipleSelectionData(key, rowData) {
    if (enableToggleMode && selectedRowsKeys.includes(key)) {
      return getMultipleSelectionToggleData(key, rowData);
    }

    if (!selectedRowsKeys.includes(key)) {
      const data = [...selectedRowsData, rowData];
      setSelectedRowsData(data);
      setSelectedRowsKeys([...selectedRowsKeys, key]);
      return data;
    }
    return [...selectedRowsData];
  }

  function handleClick(key, rowData) {
    if (onChange && modes.includes(updatedMode)) {
      let data;
      if (updatedMode === "MULTIPLE") {
        data = getMultipleSelectionData(key, rowData);
      } else {
        data = getSingleSelection(key, rowData);
      }
      onChange(data);
    }
  }

  return (
    <tbody className={styles(selectionMode)}>
      {Array.isArray(dataRows) &&
        dataRows.map((rowData, index) => {
          const getKey = generateKey("tr", rowData.join(), index);
          const rowSelectedCls = selectedRowsKeys.includes(getKey)
            ? "selected-row-highlight"
            : "";
          return (
            <tr
              key={getKey}
              onClick={() => handleClick(getKey, rowData)}
              className={styles(rowSelectedCls)}
            >
              {mode === "single" && (
                <td>
                  <input type="radio" className="input-selection" checked={selectedRowsKeys.includes(getKey)} onChange={e => {}}/>
                </td>
              )}
              {mode === "multiple" && (
                <td>
                  <input type="checkbox" className="input-selection" checked={selectedRowsKeys.includes(getKey)} onChange={e => {}}/>
                </td>
              )}
              {<TableData headerLabels={headerLabels} rowData={rowData} index={index} />}
            </tr>
          );
        })}
    </tbody>
  );
}

export default TableBody;
