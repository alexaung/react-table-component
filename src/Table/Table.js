import React, { useState, useEffect } from 'react';
import TableStyles from './Table.scss';
import classNames from 'classnames/bind';
import TableHeader from './TableHeader';
import TableBody from './TableBody';

const styles = classNames.bind(TableStyles);
const ROW_HEIGHT = 56;

function Table(props) {
  const {
    title,
    headerLabels,
    dataRows,
    mode,
    maxRowsToDisplay,
    enableToggleMode = true,
    footer,
    header,
    onChange
  } = props;
  const [maxHeight, setMaxHeight] = useState(getHeight(maxRowsToDisplay));

  function getHeight(maxRows = 5) {
    return ROW_HEIGHT * (maxRows + 1);
  }

  useEffect(() => {
    setMaxHeight(getHeight(maxRowsToDisplay));
  }, [maxRowsToDisplay]);

  return (
    <div>
      {header ? header : null}
      {title && <div className={styles('table-title')}>{title}</div>}
      <div className={styles('table-wrapper')} style={{ maxHeight }}>
        <table className={styles('base-table')}>
          {Array.isArray(headerLabels) && (
            <TableHeader headerLabels={headerLabels} mode={mode} />
          )}
          {Array.isArray(dataRows) && (
            <TableBody
              headerLabels={headerLabels}
              dataRows={dataRows}
              mode={mode}
              onChange={onChange}
              enableToggleMode={Boolean(enableToggleMode)}
            />
          )}
        </table>
      </div>
      {footer ? footer : null}
    </div>
  );
}

export default Table;
