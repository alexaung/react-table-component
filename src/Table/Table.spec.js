import React from 'react';
import Table from './Table';
import { render, cleanup, fireEvent } from '@testing-library/react';

describe('<Table/>', () => {
  const headerLabels = [
    'Claim Number',
    'Date of Admission',
    'Date of Discharge',
    'Clinic/Hospital'
  ];
  const dataRows = [
    ['test-id', '12-03-2021', '26-03-2021', 'PARKWAY EAST'],
    ['CBGDC21000386-00', '13-03-2021', '25-03-2021', 'RESTRUCTURED HOSPITAL'],
    ['CBGDC21000386-00', '14-03-2021', '24-03-2021', 'PARKWAY EAST'],
    ['CBGDC21000386-00', '15-03-2021', '23-03-2021', 'SUNWAY'],
    ['CBGDC21000386-00', '16-03-2021', '22-03-2021', 'KEPPAL'],
    ['CBGDC21000386-00', '17-03-2021', '21-03-2021', 'PARKWAY EAST'],
    ['CBGDC21000386-00', '18-03-2021', '20-03-2021', 'PARKWAY EAST']
  ];
  afterEach(cleanup);

  it('Should render basic component', () => {
    const { container } = render(
      <Table
        headerLabels={headerLabels}
        dataRows={dataRows}
        mode="single"
        title="test title"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should invoke onChange callback', () => {
    const onChange = jest.fn();
    const { getByText } = render(
      <Table
        headerLabels={headerLabels}
        dataRows={dataRows}
        mode="single"
        title="test title"
        onChange={onChange}
      />
    );
    fireEvent.click(getByText('test-id'));
    expect(onChange).toHaveBeenCalled();
  });
});
