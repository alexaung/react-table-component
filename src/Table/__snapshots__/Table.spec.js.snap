// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> Should render basic component 1`] = `
<div>
  <div
    class="table-title"
  >
    test title
  </div>
  <div
    class="table-wrapper"
    style="max-height: 336px;"
  >
    <table
      class="base-table"
    >
      <thead>
        <tr>
          <th />
          <th
            scope="col"
          >
            Claim Number
          </th>
          <th
            scope="col"
          >
            Date of Admission
          </th>
          <th
            scope="col"
          >
            Date of Discharge
          </th>
          <th
            scope="col"
          >
            Clinic/Hospital
          </th>
        </tr>
      </thead>
      <tbody
        class="enable-table-hover"
      >
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            test-id
          </td>
          <td>
            12-03-2021
          </td>
          <td>
            26-03-2021
          </td>
          <td>
            PARKWAY EAST
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            CBGDC21000386-00
          </td>
          <td>
            13-03-2021
          </td>
          <td>
            25-03-2021
          </td>
          <td>
            RESTRUCTURED HOSPITAL
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            CBGDC21000386-00
          </td>
          <td>
            14-03-2021
          </td>
          <td>
            24-03-2021
          </td>
          <td>
            PARKWAY EAST
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            CBGDC21000386-00
          </td>
          <td>
            15-03-2021
          </td>
          <td>
            23-03-2021
          </td>
          <td>
            SUNWAY
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            CBGDC21000386-00
          </td>
          <td>
            16-03-2021
          </td>
          <td>
            22-03-2021
          </td>
          <td>
            KEPPAL
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            CBGDC21000386-00
          </td>
          <td>
            17-03-2021
          </td>
          <td>
            21-03-2021
          </td>
          <td>
            PARKWAY EAST
          </td>
        </tr>
        <tr
          class=""
        >
          <td>
            <input
              class="input-selection"
              type="radio"
            />
          </td>
          <td>
            CBGDC21000386-00
          </td>
          <td>
            18-03-2021
          </td>
          <td>
            20-03-2021
          </td>
          <td>
            PARKWAY EAST
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
